src/printf.h

```diff
@@ -9,4 +11,30 @@ struct Field;
 
 std::string verify_format_string(const std::string &fmt, std::vector<Field> args);
 
+class IPrintable
+{
+public:
+  virtual ~IPrintable() { };
+  virtual uint64_t value() = 0;
+};
+
+class PrintableString : public virtual IPrintable
+{
+public:
+  PrintableString(std::string value) : _value(value) { }
+  uint64_t value();
+private:
+  std::string _value;
```

Comment:
Could you change these to `value_` (underscore after) please? Just want to keep a consistent style across the project